hazelcast/src/main/java/com/hazelcast/internal/util/ThreadLocalRandomProvider.java

```diff
@@ -38,7 +40,8 @@ private ThreadLocalRandomProvider() {
     public static Random get() {
         Random random = THREAD_LOCAL_RANDOM.get();
         if (random == null) {
-            random = new Random();
+            long seed = SEED_GENERATOR.nextLong();
```

Comment:
Random is just thread-safe since Java 7, isn't it? May this cause issues when this method is called from different threads?